amazon-import.user.js

```diff
@@ -0,0 +1,291 @@
+    margin:0.5em;
+      color: white;
+      background-color: #04aa6d;
+      border:none;
+
+  }
+`,
+  `
+  #bb-cancel{  
+    background-color:red;
+
+  }
+`,
+  `
+  .bb-btn:hover{
+    cursor:pointer;
+    background-color: #eb743b;
+    color: white;
+  }
+`,
+  `
+  .bb-finput{
+    margin-bottom:0.4em
+    }
+
+`,
+  `
+  .bb-form{
+    display:flex;
+     flex-direction:column;
+      justify-content:center;
+      width:min-content;
+      padding:0.8em;
+      font-family:'mono', sans-serif;
+      
+    }
+
+`,
+  `
+  .bb-flabel {
+    margin-bottom:0.5em;
+     margin-top:0.1em;
+     font-size:1rem;
+     font-weight:700;
+   }
+
+`,
+  `
+
+  .bb-container{
+    border:2px solid black;
+      border-radius:10px;
+      display:flex;
+     flex-direction:column;
+      justify-content:center;
+      width:min-content;
+      margin:1em;
+    }
+`,
+  `
+  .bb-h3{
+    padding:0.4em;
+      background-color:#754e37;
+      margin:0;
+      border-radius:8px;
+      text-align:center;
+      color:white;
+      
+    }
+`,
+];
+GM_addStyle(cssRules, "bookbrainz");
+const convertToBB = {
+  cm: 10,
+  g: 1,
+  pounds: 453.6,
+  ounces: 28.3,
+  inches: 2.5,
+};
+// #productTitle Name/Sort Name
+// ul.a-spacing-none:nth-child(1) Language/Pages/Dimensions/Weight/Publisher(Date)/ISBNs
+function scrapeAmz() {
+  let name, sortName;
+  name = sortName = document.getElementById("productTitle").innerText;
+  const prodDetails = document.querySelector("ul.a-spacing-none:nth-child(1)");
+  const subtitleEl = document.getElementById("productSubtitle");
+  let [format, date, ...res] = subtitleEl.innerText.split(" – ");
+  const prodDetailsMap = {
+    publisher: "Publisher",
+    language: "Language",
+    pages: format.trim(),
+    isbn10: "ISBN-10",
+    isbn13: "ISBN-13",
+    weight: "Item Weight",
+    dimensions: "Dimensions",
+  };
+  const reverseProdDetailsMap = Object.fromEntries(
+    Object.entries(prodDetailsMap).map((b) => b.reverse())
+  );
+  let key, value;
+  res = {};
+  for (
+    let index = 0;
+    index < Object.keys(prodDetails.children).length;
+    index++
+  ) {
+    let ls = prodDetails.children[index].innerText.split(":");
+    key = ls[0]
+      .replace(/\u200f/g, "")
+      .replace(/\u200e/g, "")
+      .trim();
+    value = ls[1]
+      .replace(/\u200f/g, "")
+      .replace(/\u200e/g, "")
+      .trim();
+    if (reverseProdDetailsMap[key]) {
+      res[reverseProdDetailsMap[key]] = value;
+    }
+  }
+  let [height, width, depth] = res.dimensions?.split("x");
+  let lenghtToSIkey = depth.match(/[A-Za-z]+/gi)[0];
+  let wtToSIkey = res.weight?.match(/[A-Za-z]+/gi)[0];
+  let publisher;
+  publisher = res.publisher?.split(";")[0]?.split("(")[0]?.trim();
+  date = new Date(date.replace(".", "")); // temporary fix for unsupported dates like `20 Oct. 2021`
+  date = [
+    date.getFullYear(),
+    ((date.getMonth() < 10) ? "0" : "") + `${date.getMonth() + 1}`,
+    date.getDate(),
+  ].join("-");
```

Comment:
Sorry, I forgot to pad the date (was not necessary for my example). The day also has to be padded, so I suggest to use the following (year is already long enough, so it will be left as is by the mapper function):
```suggestion
date = [date.getFullYear(), date.getMonth() + 1, date.getDate()]
    .map((component) => String(component).padStart(2, '0'))
    .join('-');
```